regression/cbmc/Pointer_byte_extract5/main.i

```diff
@@ -11,13 +11,13 @@ typedef union
 typedef struct
 {
   int Count;
-  Union List[1];
+  Union List[0];
```

Comment:
:question: Why does this change? The change should go in a separate commit with a why

Reply:
This was undefined behaviour (and the changed in malloc exposed that CBMC does not really support that). It now has separate commit with explanation.

Reply:
I find it rather surprising that the changes in this PR expose the lack of support for this C90-hack. What exactly was going wrong here?

Reply:
Why this particular PR exposes the problem is a bit more complicated. But the problem itself can be easily reproduced, e.g. (CMBC reports the assertion as failing):
```
#include <malloc.h>
#include <assert.h>

struct Foo
{
  int i;
  char data[1];
};

int main()
{
  struct Foo* foo = malloc(sizeof(struct Foo) + sizeof(char));
  foo->data[1]='b'; // set data[1]
  assert(foo->data[1]=='b'); // check data[1] -- should succeed

  return 0;
}
```